app/api/csv.py

```diff
@@ -0,0 +1,43 @@
+import re
+import six
+import csvx
+from io import StringIO
+from collections import OrderedDict as od
+
+
+def generate_brief_responses_csv(brief, responses):
+    def csv_cell_sanitize(text):
+        return re.sub(r"^(=|\+|-|@|!|\|{|}|\[|\]|<|,)+", '', unicode(text).strip())
+
+    def row(r):
+        answers = od()
+
+        ess_req_names = brief.data.get('essentialRequirements', [])
+        nth_req_names = brief.data.get('niceToHaveRequirements', [])
+        ess_responses = r.data.get('essentialRequirements', ['']*len(ess_req_names))
+        nth_responses = [r.data.get('niceToHaveRequirements')[i] if
+                         i < len(r.data.get('niceToHaveRequirements', [])) else
+                         '' for i in range(len(nth_req_names))]
+
+        answers.update({'Supplier': r.supplier.name})
+        answers.update({'Contact': r.data.get('respondToEmailAddress', 'UNKNOWN')})
+        if brief.lot.slug == 'digital-professionals':
+            answers.update({'Specialist Name': r.data.get('specialistName', 'UNKNOWN')})
+        answers.update({'Availability Date': r.data.get('availability', 'UNKNOWN')})
+        answers.update({'Day rate': r.data.get('dayRate', '')})
+        answers.update(zip(ess_req_names, ess_responses))
+        answers.update(zip(nth_req_names, nth_responses))
+
+        for k, v in answers.items():
+            answers[k] = csv_cell_sanitize(v)
+
+        return answers
+
+    rows = [row(_) for _ in responses]
```

Comment:
i rarely say this, but perhaps you should add a comment - this is confusing

Reply:
This is basically a direct copy from buyer FE's CSV code but I agree, it could probably be made to be simpler.